rotate.go

```diff
@@ -602,6 +620,44 @@ func (r *Rotator) rollback(ctx context.Context, creds db.NewPassword, rotationSt
 	return errRotationFailed // always return this error
 }
 
+// checks that secret have been replicated to all replica regions
+// this is necessary between multiple calls of UpdateSecretVersionStage
+// to guard against arace condition in AWS that leaves secret replication
+// stuck indefinitely.
+func (r *Rotator) checkSecretReplicationStatus() error {
+	log.Println("checking secret replication status")
+	waitDuration := DEFAULT_REPLICATION_WAIT_DURATION
+	retries := DEFAULT_REPLICATION_WAIT_RETRIES
+	if r.replicationWaitRetries > 0 {
+		retries = r.replicationWaitRetries
+	}
+	if r.replicationWaitDuration > 0 {
+		waitDuration = r.replicationWaitDuration
+	}
+
+	for i := 0; i < retries; i++ {
+		secret, err := r.sm.DescribeSecret(&secretsmanager.DescribeSecretInput{
+			SecretId: aws.String(r.secretId),
+		})
+		if err != nil {
+			return err
+		}
+		replicationSyncComplete := true
+		for _, status := range secret.ReplicationStatus {
+			if *status.Status != secretsmanager.StatusTypeInSync {
+				replicationSyncComplete = false
+				log.Printf("replication status still in (%v) in region (%v) expecting (%v)\n", *status.Status, status.Region, secretsmanager.StatusTypeInSync)
+			}
+		}
+		if replicationSyncComplete {
+			log.Println("secret replication sync completed successfully")
+			return nil
+		}
+		time.Sleep(waitDuration)
+	}
+	return fmt.Errorf("secret replication sync is stuck")
```

Comment:
Let's make this more technical because "stuck" can mean anything. Something like:
> timeout waiting for replication StatusTypeInSync = true
Or reword to whatever code is actually looking at (and not seeing) in the API response.

Reply:
updated to 

_timeout waiting for replication StatusTypeInSync = true_